add presets for DeepSeek R1 models

kallewoof opened the pull request:
This is probably correct, but may need more testing.

## Checklist:
* [x]  I have read the [Contributing guidelines](https://github.com/SillyTavern/SillyTavern/blob/release/CONTRIBUTING.md).

kallewoof committed:
add presets for DeepSeek R1 models

cloak1505 commented:
V2.5 was the first to use these instruct sequences so should we call this DeepSeek-V2.5?

Also IMO new presets should default `"names_behavior": "force"` (Groups and Past Personas, equivalent to Default in Chat Completion).

kallewoof commented:
> V2.5 was the first to use these instruct sequences so should we call this DeepSeek-V2.5?

Thanks, didn't realize. Calling it V2.5 might make people unsure if it applies to R1 as well, but I'm neutral on either name.
 
> Also IMO new presets should default `"names_behavior": "force"` (Groups and Past Personas, equivalent to Default in Chat Completion).

OK, that works for me.

cloak1505 commented:
`"system_same_as_user": true` since no sequences.

kallewoof committed:
fixes

kallewoof commented:
> `"system_same_as_user": true` since no sequences.

Won't that make it output the initial system message as <|User|> then? The template does support system messages, they just don't have a prefix/suffix.

cloak1505 commented:
> Won't that make it output the initial system message as <|User|> then?

No, beginning of context template in TC doesn't get sequences applied, hence why you see other context templates with a sequence written in.

Edit: I don't know how to interpret chat_template in tokenizer_config.json. ~~Is it really documented use to have non-sequenced prompts between user and assistant turns i.e. outside of beginning?~~ Edit 2: Oh. Below shows they aren't expecting mid-chat system role.

kallewoof commented:
> > Won't that make it output the initial system message as <|User|> then?
> 
> No, beginning of context template in TC doesn't get sequences applied, hence why you see other context templates with a sequence written in.

Oh, all right. Set to true.

> Edit: I don't know how to interpret chat_template in tokenizer_config.json. Is it really documented use to have non-sequenced prompts between user and assistant turns i.e. outside of beginning?

I'm not sure what you are referring to. Here  it is without tool stuff and some whitespace:
```jinja
{% if not add_generation_prompt is defined %}
    {% set add_generation_prompt = false %}
{% endif %}
{% set ns = namespace(is_first=false, is_tool=false, is_output_first=true, system_prompt='') %}
{%- for message in messages %}
    {%- if message['role'] == 'system' %}
        {% set ns.system_prompt = message['content'] %}
    {%- endif %}
{%- endfor %}
{{bos_token}}
{{ns.system_prompt}}
{%- for message in messages %}
    {%- if message['role'] == 'user' %}
        {{'<｜User｜>' + message['content']}}
    {%- endif %}
    {%- if message['role'] == 'assistant' and message['content'] is not none %}
        {% set content = message['content'] %}
        {% if '</think>' in content %}
            {% set content = content.split('</think>')[-1] %}
        {% endif %}
        {{'<｜Assistant｜>' + content + '<｜end▁of▁sentence｜>'}}
    {%- endif %}
{%- endfor -%}
{% if add_generation_prompt and not ns.is_tool %}{{'<｜Assistant｜>'}}{% endif %}
```
Rewritten in python:
```py
for message in messages:
    if message['role'] == 'system':
        system_prompt = message['content'] # last entry overrides if multiple
print(bos_token + system_prompt)
for message in messages:
     if message['role'] == 'user';
        print("<｜User｜>" + message['content'])
    if message['role'] == 'assistant' and message['content'] is not None:
        content = message['content']
        if '</think>' in content:
            content = content.split("</think>")[-1]  # rip out old thought entries
        print('<｜Assistant｜>' + content + '<｜end▁of▁sentence｜>')
    # ignore all other roles, including system
if add_generation_prompt:
    print('<｜Assistant｜>')
````

kallewoof committed:
system same as user

Vibecoder9000 commented:
Suggestion, change the name to Deepseek-V2.5/V3/R1. We don't need to truncate it that badly. It would help a lot with understanding.
